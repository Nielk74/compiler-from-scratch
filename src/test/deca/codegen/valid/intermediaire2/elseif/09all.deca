// Description:
// Test de plusieurs 3 if avec plusieur else if
// Test le else de la 1ere condition
// Test le if de la 2eme codition
// Test le else if de la 3eme condition
//
// Resultats:
//  !(421 == 100) && !(421 < 421) && !(421 > 421)
// 421 == 421
// !(4 == 100) && 4 < 421
//
// Historique:
//    crÃ©e le 10/06/2022
{
    int secret, user;
    secret = 421;
    user = 421;
    if (user == 100) {
        println("421 == 100");
    }else if (user < secret) {
        println("!(421 == 100) && 421 < 421");
    } else if (user > secret) {
        println("!(421 == 100) && !(421 < 421) && 421 > 421");
    } else{
        println("!(421 == 100) && !(421 < 421) && !(421 > 421)");
    } 
    
    if (user == secret) {
        println("421 == 421");
    }else if (user > secret) {
        println("!(421 == 421) && 421 > 421");
    } else if (user < secret) {
        println("!(421 == 421) && !(421 > 421) && 421 < 421");
    }

    user = 4;
    if (user == 100) {
        println("4 == 100");
    }else if (user < secret) {
        println("!(4 == 100) && 4 < 421");
    } else if (user > secret) {
        println("!(4 == 100) && !(4 < 421) && 4 > 421");
    } else{
        println("!(4 == 100) && !(4 < 421) && !(4 > 421)");
    } 
}