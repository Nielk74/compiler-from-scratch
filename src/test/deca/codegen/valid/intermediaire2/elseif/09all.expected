!(421 == 100) && !(421 < 421) && !(421 > 421)
421 == 421
!(4 == 100) && 4 < 421
